src/api/rpc_api/subscribe_events.rs

```diff
@@ -75,11 +75,7 @@ where
 				// Check for failed xt and return as Dispatch Error in case we find one.
 				// Careful - this reports the first one encountered. This event may belong to another extrinsic
 				// than the one that is being waited for.
-				if extrinsic_has_failed(&event_details) {
-					let dispatch_error =
-						DispatchError::decode_from(event_details.field_bytes(), self.metadata());
-					return Err(Error::Dispatch(dispatch_error))
-				}
+				event_details.check_failed()?;
```

Comment:
I am uncertain if I agree with the naming. I propose a few here:

* check_if_failed
* xt_successful
* xt_failed

What do you think?

Reply:
I was thinking about the first one as well. Let me adapt.